tools/pkg/platforms/centos7/files/build/script.sh

```diff
@@ -1,46 +1,54 @@
 #!/usr/bin/env bash
-COMMIT=$1
-REV=$2
 set -e
 
-HOSTNAME=`hostname`
-ARCH="x86_64"
+git_ref=$1
+revision=$2
+repo_url=$3
+specfile_down_erl_vsn=$4
+specfile_up_erl_vsn=$5
 
-MONGOOSEIM_REPO=${MONGOOSEIM_REPO:-https://github.com/esl/mongooseim.git}
+arch="x86_64"
+package_name_arch="amd64"
 
-wget http://packages.erlang-solutions.com/erlang-solutions-1.0-1.noarch.rpm
-sudo rpm -Uvh --replacepkgs erlang-solutions-1.0-1.noarch.rpm
-
-git clone ${MONGOOSEIM_REPO} mongooseim
+git clone "$repo_url" mongooseim
 cd mongooseim
-git checkout ${COMMIT}
-
-VER=$(cat VERSION)
-
-
-export ERL_TOP=`pwd`
-sudo groupadd mongooseim
-sudo adduser -g mongooseim mongooseim
-mkdir -p rpmbuild/{RPMS,SOURCES,SPECS,SRPMS,BUILD,BUILDROOT/mongooseim-${VER}-${REV}.${ARCH}/}
-chmod 777 -R rpmbuild
-cp ../mongooseim.spec rpmbuild/SPECS/mongooseim.spec
-
-sed -i "s#@ARCH@#${ARCH}#" rpmbuild/SPECS/mongooseim.spec
-
-sudo yum-builddep rpmbuild/SPECS/mongooseim.spec -y
-sudo yum erase rpm-build -y
-sudo yum install rpm-build -y
-
-cp ../rpmmacros.in            $HOME/.rpmmacros
-sed -i "s#@ARCH@#${ARCH}#"    $HOME/.rpmmacros
-sed -i "s#@VSN@#${VER}#"      $HOME/.rpmmacros
-sed -i "s#@REVISION@#${REV}#" $HOME/.rpmmacros
-
-./tools/configure with-all user=mongooseim prefix=/ system=yes
-sed -i "s#PREFIX=\"/\"#PREFIX=\"rpmbuild/BUILDROOT/mongooseim-${VER}-${REV}.${ARCH}/\"#" configure.out
-source configure.out
-make install
-
-rpmbuild -bb rpmbuild/SPECS/mongooseim.spec
-
-mv rpmbuild/RPMS/${ARCH}/mongooseim-${VER}-${REV}.${ARCH}.rpm ../mongooseim-${VER}-${REV}~${HOSTNAME}.${ARCH}.rpm
+git checkout "$git_ref"
+
+version=$(cat VERSION)
+commit_sha=$(git rev-parse --short HEAD)
+
+# Adjust package revision to requirements:
+# https://twiki.cern.ch/twiki/bin/view/Main/RPMAndDebVersioning
+if git show-ref --verify "refs/tags/$git_ref" &>/dev/null && $version == "$git_ref"; then
+    package_revision="${revision}"
+elif git show-ref --verify "refs/heads/$git_ref" &>/dev/null; then
+    package_revision="${revision}.${git_ref}.${commit_sha}"
+else
+    package_revision="${revision}.${commit_sha}"
+fi
+package_version="${version}-${package_revision}"
```

Comment:
Can we use the script `tools/generate_vsn.sh` to calculate the package version? This scripts returns:
* building from a tag, it will give you the tag only `3.6.0`
* building from any other commit it will give you the recent version with extra information like number of commits since the tag and the recent commit sha: `3.6.0-30-g764a681a1`
* building from an archive (no git files), it will give you what's inside the `VERSION` file.